src/pyubx2/ubxreader.py

```diff
@@ -138,7 +141,7 @@ def read(self) -> tuple:
                 byte2 = self._read_bytes(1)
                 bytehdr = byte1 + byte2
                 # if it's a UBX message (b'\xb5\x62')
-                if bytehdr == UBX_HDR:
+                if bytehdr == UBX_HDR and self._protfilter & UBX_PROTOCOL:

```

Comment:
Not clear why this is needed. If the protfilters are set, you shouldn't be receiving the filtered protocols at the client, so there should be no 'noise' as such. The reason for parsing them even if the protocol is filtered is simply to remove them from the input buffer as efficiently as possible (i.e. with the minimum i/o).

For your particular use case, I think a better approach might be to suppress the unwanted protocols at the port level before doing the firmware upgrade and then enable them again afterwards e.g. with a CFG-PRT command. This also reduces the latency on any ACKs.